# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/RemoteBrowser.h
# Commit: 1663719e2aa5
# Full Hash: 1663719e2aa5b6f77e32398c6d6fcb4cb3c999ed
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 15:55:11
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 16 - Cycle collect RemoteBrowser and support weak references in BrowserHost. r=nika
#   
#   BrowserParent is cycle collected and supported weak references, so this commit adds support
#   for these things to BrowserHost.
#   
# ==============================================================================

diff -r 53663cad8890 -r 1663719e2aa5 dom/ipc/RemoteBrowser.h
--- a/dom/ipc/RemoteBrowser.h	Wed May 08 16:12:26 2019 -0500
+++ b/dom/ipc/RemoteBrowser.h	Tue May 14 16:28:10 2019 -0500
@@ -11,6 +11,7 @@
 #include "mozilla/dom/ipc/IdType.h"
 #include "mozilla/layers/LayersTypes.h"
 #include "nsILoadContext.h"
+#include "nsISupports.h"
 #include "nsISupportsImpl.h"
 #include "nsIURI.h"
 #include "nsRect.h"
@@ -33,10 +34,8 @@
  * nsFrameLoader will use BrowserHost, and a content process nsFrameLoader will
  * use BrowserBridgeHost.
  */
-class RemoteBrowser {
+class RemoteBrowser : public nsISupports {
  public:
-  NS_INLINE_DECL_PURE_VIRTUAL_REFCOUNTING
-
   typedef mozilla::layers::LayersId LayersId;
 
   // Try to cast this RemoteBrowser to a BrowserHost, may return null
